docs/testing-strategy.md

```diff
@@ -0,0 +1,152 @@
+corresponding tests in the spec/lib directory.
+
+Directories within spec that don't contain Ruby unit tests are as follows:
+
+- factories - contains [FactoryBot factories][factory-bot], which are a fixture
+  alternative;
+- features - contains [feature tests](#feature-tests), which exercises
+  application features from a user perspective;
+- javascripts - contains [unit tests](#unit-tests) for JavaScript files
+  and configuration and helper files for Jasmine;
+- requests - contains [request tests](#request-tests), which test the HTTP API
+  of the application;
+- support - contains helper files for shared test methods and logic;
+- views - contains [view tests](#view-tests), which are used to test
+  [Rails views][rails-views] output expected HTML in a given scenario.
+
+## Unit tests
+
+The purpose of these tests is to test individual units of the system in
+isolation. They are intended to provide exhaustive tests of the code paths
+through a particular class or module through its public interface. As per
+the [test pyramid](https://martinfowler.com/bliki/TestPyramid.html) approach
+to software testing, these should provide the majority of test coverage.
+
+Characteristics of unit tests:
+
+- they should be concerned with the logic within the class/module being tested,
+  and not test logic defined elsewhere;
+- they may mock dependent objects and/or assert that particular external
+  methods are called;
+- they should test all code paths through a class/module;
+- they should test that side effects intended by the code occur, for example writing
+  to the database or making API calls.
+
+## Feature tests
+
+Feature tests are used to assert that Content Publisher behaves correctly from
+the perspective of someone using the application. This ensures we're testing
+the system in a way representative of expected user behaviour. This makes the
+tests some of the most valuable of the application, however they are slower
+than other tests to run and can be difficult to debug. Therefore they are not
+intended to exhaustively test all the scenarios that can occur.
+
+Characteristics of feature tests:
+
+- they should test across multiple requests - consider a
+  [request test](#request-tests) to test a single request;
+- they should navigate the application using the same means as a user, for
+  example clicking links or submitting forms;
+- they shouldn't test dead ends in a user flow (such as validation or
+  permission issues); these are simpler tested in
+  [request tests](#request-tests);
+- incidental behaviour (for example flash messages) shouldn't be tested, this
+  can be tested at a higher granularity in [unit](#unit-tests) or
+  [request](#request-tests) tests;
+- they shouldn't test the effects a test has on the database, only user visible
+  signs of success should be asserted on;
+- they shouldn't mock application code;
+- there may be mocking and assertions that particular API calls are made to
```

Comment:
How about 'they may mock and/or assert that particular API calls are made to external services;', it follows the wording flow of line 65 which I think is an improvement - https://github.com/alphagov/content-publisher/pull/1590/commits/b55b02737011b70e64629e2090964458c3b96158#diff-18a0a75d23cfa0c005b7d73b8c2477ddR65.

Reply:
Good shout.